### PR TITLE
[ATL-1107][ATL-1045] Verify/Upload Buttons Hover & "running" style

### PR DESCRIPTION
**Pull request by fstasi**
Creating another PR as I accidentally merged #190 

This PR is an amend to the previous one, containing fixes after @kittaakos review

I Will **squash merge** this PR into main once the review is over, in order to have a single commit on the main branch

### CODE REVIEW

File: arduino-ide-extension/src/browser/contributions/upload-sketch.ts [hunk 1]
```diff
@@ -69,6 +75,16 @@ export class UploadSketch extends SketchContribution {
     }
 
     async uploadSketch(usingProgrammer: boolean = false): Promise<void> {
+        
+        // even with buttons disabled, better to double check if an upload is already in progress
+        if (this.uploadInProgress) {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (kittaakos) [reply arduino-ide-extension/src/browser/contributions/upload-sketch.ts:82]: > but the check whether or not continue with the action should be inside the action handler itself, in my opinion.

It is there; you added the `isEnabled` method: https://github.com/arduino/arduino-ide/pull/197/files#diff-5c246e36e23a59755c9cc7d5a116eac5be34d188c0eb681f8c449915cbfe9aacR30.

In theory, if the command handler is not enabled, the command should not run. I am fine with this approach. I just wanted to know if you did hit an error or why did you add it.

**Comment by kittaakos** [comment]
@fstasi, if the PR is ready for review, please quash the commits, and I look into it. Thank you!

**Comment by fstasi** [comment]
> @fstasi, if the PR is ready for review, please quash the commits, and I look into it. Thank you!

@kittaakos sqashed and ready for review :)

**Comment by kittaakos** [comment]
I have a few remarks on the behavior:

 - The toolbar color is not yellow when executing the command and hovering the toolbar. See the screencast. Once I have clicked on it, I have to un-hover the toolbar to see that it's yellow.

screencast 2021-03-15 09-42-38
In the Java IDE, the toolbar is yellow, when running and hover. This does not happen with IDE2.

 - I have  noticed another issue, the toolbar hover has a different color:
Java IDE:
screencast 2021-03-15 10-46-38

IDE2:
screencast 2021-03-15 10-47-41

 - There is a pixel mismatch or what, one can clearly see how the toolbar shifts to the right a bit when the command is in process, then goes back.
screencast 2021-03-15 10-49-06

**Comment by kittaakos** [comment]
I have restarted the build, it was failing due to some issues out of the context of the project:
```
    gyp http 500 https://electronjs.org/headers/v9.4.3/node-v9.4.3-headers.tar.gz
    gyp WARN install got an error, rolling back install
    gyp ERR! configure error 
    gyp ERR! stack Error: 500 response downloading https://electronjs.org/headers/v9.4.3/node-v9.4.3-headers.tar.gz
```
HTTP 5xx is a server error, there isn't much we could do. Hopefully, we get back the green build 🤞

**Review by ubidefeo — CHANGES_REQUESTED** [review]
the behaviour is correct, but disabled buttons (such as the debug button) should not receive a hover when they are disabled.

steps to reproduce:
- open a sketch
-  select Arduino UNO (does not support debugging)
- ensure the button is greyed out
- hover 
- verify it changes colour

**Comment by fstasi** [comment]
> the behaviour is correct, but disabled buttons (such as the debug button) should not receive a hover when they are disabled.
> 
> steps to reproduce:
> 
>     * open a sketch
> 
>     * select Arduino UNO (does not support debugging)
> 
>     * ensure the button is greyed out
> 
>     * hover
> 
>     * verify it changes colour

@ubidefeo fixed, thanks for pointing that out. I'm glad I had a spare very old arduino uno to test it 🧪 

let me know if we are good to go

**Review by ubidefeo — APPROVED** [review]
LGTM

**Pull request merged**